dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -92,24 +120,318 @@ public ContainerResource(final WebResource webResource,
+            .findFirst()
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find container structure template:" + contentletId);
+            });
+
+
+
+        // defensive copy
+        container = (Container) BeanUtils.cloneBean(container);
+        container.setPreLoop(null);
+        container.setPostLoop(null);
+
+
+        VelocityUtil.getEngine()
+            .evaluate(context, out, this.getClass()
+                .getName(), ContainerServices.buildVelocity(container, containerIdentifier, false));
+
+        Map<String, String> response = new HashMap<>();
+        response.put("render", out.toString());
+
+        return Response.ok(response)
+            .build();
+
+    }
+
+
+
+    @POST
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("add/{containerId}/content/{contentletId}/uid/{uid}/order/{order}")
+    public final Response addContentToContainer(@Context final HttpServletRequest req, @Context final HttpServletResponse res,
+            @PathParam("containerId") final String containerId, @PathParam("contentletId") final String contentletId,
+            @QueryParam("order") final int order, @PathParam("uid") final String uid) throws DotDataException,
+            DotSecurityException, ParseErrorException, MethodInvocationException, ResourceNotFoundException, IOException,
+            IllegalAccessException, InstantiationException, InvocationTargetException, NoSuchMethodException {
+
+
+        final InitDataObject initData = webResource.init(true, req, true);
+        final User user = initData.getUser();
+        final PageMode mode = PageMode.get(req);
+        final Language id = WebAPILocator.getLanguageWebAPI()
+            .getLanguage(req);
+
+        ShortyId contentShorty = APILocator.getShortyAPI()
+            .getShorty(contentletId)
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find contentlet:" + contentletId);
+            });
+        final Contentlet contentlet = (contentShorty.subType == ShortType.CONTENTLET) ? APILocator.getContentletAPI()
+            .find(contentShorty.longId, user, mode.showLive)
+                : APILocator.getContentletAPI()
+                    .findContentletByIdentifier(contentShorty.longId, mode == PageMode.ANON, id.getId(), user,
+                            mode == PageMode.ANON);
+
+
+
+        ShortyId containerShorty = APILocator.getShortyAPI()
+            .getShorty(containerId)
+            .orElseGet(() -> {
+                throw new ResourceNotFoundException("Can't find Container:" + containerId);
+            });
+        Container container = (containerShorty.subType == ShortType.CONTAINER) ? APILocator.getContainerAPI()
+            .find(containerId, user, mode == PageMode.ANON)
+                : (mode.showLive) ? (Container) APILocator.getVersionableAPI()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)